src/openeo_aggregator/metadata/merging.py

```diff
@@ -0,0 +1,327 @@
+                        default_keys = {
+                            "default": None, "optional": False, "deprecated": False, "experimental": False
+                        }
+                        values_changed = diff.get('values_changed', None)
+                        if values_changed:
+                            paths_to_remove = []
+                            for path, changes in values_changed.items():
+                                if type(changes) != dict:
+                                    continue
+                                new_value, old_value = changes.get("new_value", None), changes.get("old_value", None)
+                                if type(new_value) == type(old_value) == dict:
+                                    for key, value in default_keys.items():
+                                        if key not in new_value:
+                                            new_value[key] = value
+                                        if key not in old_value:
+                                            old_value[key] = value
+                                if new_value == old_value:
+                                    paths_to_remove.append(path)
+                            for path in paths_to_remove:
+                                del values_changed[path]
+                            if not values_changed:
+                                del diff['values_changed']
+                        # If there are still differences in the diff dict, log them.
+                        if diff:
+                            report(
+                                f"{process_id}: schema of {parameter_name} parameter is different on "
+                                f"these two groups of backends: {first_bid} and {[second_bid]}. Diff: {diff}",
+                                process_id=process_id,
+                                level="warning",
+                            )
+                            continue
+                # Compare the return schema of the process.
+                first_returns_schema = first_process.get("returns", {}).get("schema", None)
+                second_returns_schema = second_process.get("returns", {}).get("schema", None)
+                if first_returns_schema != second_returns_schema:
+                    diff = DeepDiff(
+                        first_returns_schema,
+                        second_returns_schema,
+                        ignore_order=True,
+                        exclude_regex_paths="(deprecated)|(description)|(experimental)",
+                    ).to_dict()
+                    report(
+                        f'"return" key is different on these two groups of backends: '
+                        f"{first_bid} and {[second_bid]}. Diff: {diff}",
+                        process_id=process_id,
+                        level="warning",
+                    )
+                    continue
+                first_process["supported_by"].append(bid)
```

Comment:
The use of `continue` above skips `first_process["supported_by"].append(bid)`, which is not always necessary

Reply:
This means that if the return key is different between the current list of processes and the one we want to append, then it does not get appended. Should I change this?

Reply:
I think backends should be added to this "supported_by" even when there are differences

at the moment the following simple test fails: https://github.com/Open-EO/openeo-aggregator/pull/77/commits/585d9c5cb5aaa8f22b09f720ed185310f56ca0ab#diff-ca9fe0c426dcb5d6000c48631b235c6448b0f99cb66ab4a3f8ffbf986e490c84

Reply:
resolved now